src/inputs/select-input.jsx

```diff
@@ -41,25 +41,36 @@ class SelectInput extends React.Component {
   };
 
   selectOpt = val => {
-    this.props.onChange(this.props.formId, this.props.name, val)
+    this.focusOnTypeAhead(null, true, true)
     this.setState({
+      focusedOption: null,
       hasUsedPresentationElements: true,
       showList: false,
       sortBy: null,
       sortedOpts: this.props.options
     })
-    this.focusOnTypeAhead(null, true)
+    this.props.onChange(this.props.formId, this.props.name, val)
   };
 
-  focusOnTypeAhead = (e, override = false) => {
+  focusOnTypeAhead = (e, override = false, dontShowList = false) => {
```

Comment:
This probably doesn't need to be commented on, but it may be a little clearer as `showList = true`